### PR TITLE
FSF-7142 | Pass through filters

### PR DESCRIPTION
**Pull request by Flopes55**
Added acceleration and skidpad filters + tests

### CODE REVIEW

File: src/perception/src/fov_trimming/acceleration_trimming.cpp [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    double distance = std::sqrt((point.x) * (point.x) + point.y * point.y);
+
+    // Calculate the angle of point in the XY plane in deegres
+    double angle = std::atan2(point.y, point.x) * 180 / M_PI;
+
+    if (point.z >= pc_rlidar_max_height || distance > pc_max_range ||
+        distance <= pc_min_range) {  // Ignore points too close, too far, or higher than a
+                                     // defined height relative
+      continue;                      // to the LIDAR's position
+    }
```
Comment (Davide64-dev) [inline src/perception/src/fov_trimming/acceleration_trimming.cpp:32]: I think this part should be separated from these implementations. It is just repeated. Is a code smell

**Review by Davide64-dev — CHANGES_REQUESTED** [review]

**Comment by Davide64-dev** [comment]
@Flopes55 BTW, don't put the trimming mode as a parameter, make the adapter read the mission from the topic and adapt to it dynamically

**Comment by Davide64-dev** [comment]
@Flopes55

**Comment by Davide64-dev** [comment]
@Flopes55 please solve sonarcloud

File: src/perception/src/fov_trimming/acceleration_trimming.cpp [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+  double distance = 0, angle = 0;
+  for (auto& point : cloud->points) {
+    process_point(point, distance, angle);
+
+    if (point.z >= params_.max_height - params_.lidar_height || distance > params_.max_range ||
+        distance <= params_.min_range) {  // Ignore points too close, too far, or higher than a
+                                          // defined height relative
+      continue;                           // to the LIDAR's position
+    }
```
Comment (Davide64-dev) [inline src/perception/src/fov_trimming/acceleration_trimming.cpp:23]: Please take off this. This is code redundancy

File: src/perception/src/perception/perception_node.cpp [hunk 3]
```diff
@@ -151,6 +174,12 @@ Perception::Perception(const PerceptionParameters& params)
   this->_perception_execution_time_publisher_ =
       this->create_publisher<std_msgs::msg::Float64>("/perception/execution_time", 10);
 
+  this->_master_log_subscription = this->create_subscription<custom_interfaces::msg::MasterLog>(
+      "/vehicle/master_log", rclcpp::QoS(10),
+      [this](const custom_interfaces::msg::MasterLog::SharedPtr msg) {
+        _mission_type_ = msg->mission;
+      });
+
   // Determine which adapter is being used
```
Comment (Davide64-dev) [inline src/perception/src/perception/perception_node.cpp:183]: This is good

File: src/perception/src/utils/trimming_parameters.cpp [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+
+void TrimmingParameters::set_acceleration() {
+  fov_trim_angle = 90;
+  max_range = 20.25;
+}
+void TrimmingParameters::set_skidpad() {
+  max_range = 20.25;
+  // Calculate fov_trim_angle from the given minimum distance to a cone.
+  fov_trim_angle = 90 - std::acos(1.5 / std::max(min_distance_to_cone, 1.5)) * 180 / M_PI;
+}
```
Comment (Davide64-dev) [inline src/perception/src/utils/trimming_parameters.cpp:12]: Put this as a parameter

File: src/perception/test/fov_trimming/acceleration_trimming_test.cpp [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+  params.max_height = 2.0;
+  params.lidar_height = 0;
+  params.acc_max_y = 4.0;
+  const pcl::PointCloud<pcl::PointXYZI>::Ptr cloud_ptr(
+      &cloud, NonOwningDeleter<pcl::PointCloud<pcl::PointXYZI>>());
+  const AccelerationTrimming acc_trimming(params);
+  acc_trimming.fov_trimming(cloud_ptr);
+
+  ASSERT_EQ(cloud_ptr->points.size(), 1);
+}
```
Comment (Davide64-dev) [inline src/perception/test/fov_trimming/acceleration_trimming_test.cpp:134]: Add a test for a empty point cloud

File: src/perception/test/fov_trimming/skidpad_trimming_test.cpp [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+TEST_F(SkidpadTrimmingTest, TestGeneralResult) {
+  params.min_range = 0.4;
+  params.max_height = 2.5;
+  params.min_distance_to_cone = 4.0;
+  const pcl::PointCloud<pcl::PointXYZI>::Ptr cloud_ptr(
+      &cloud, NonOwningDeleter<pcl::PointCloud<pcl::PointXYZI>>());
+  const SkidpadTrimming skidpad_trimming(params);
+  skidpad_trimming.fov_trimming(cloud_ptr);
+  ASSERT_EQ(cloud_ptr->points.size(), 1);
+}
```
Comment (Davide64-dev) [inline src/perception/test/fov_trimming/skidpad_trimming_test.cpp:97]: Please document the tests

File: src/perception/src/fov_trimming/fov_trimming.cpp [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+  const double rad = rotation * M_PI / 180.0;
+
+  // Apply rotation transformation
+  point.x = x * std::cos(rad) - y * std::sin(rad);
+  point.y = x * std::sin(rad) + y * std::cos(rad);
+
+  // Calculate distance from LIDAR on the x0y plane
+  distance = std::sqrt(point.x * point.x + point.y * point.y);
+
+  // Calculate the angle of the point in the XY plane in degrees
```
Comment (Davide64-dev) [inline src/perception/src/fov_trimming/fov_trimming.cpp:18]: Great

**Review by Davide64-dev — APPROVED** [review]
Approved
